src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -335,4 +335,15 @@ private Request matchLoansToRequest(
       .withLoan(loanMap.getOrDefault(request.getItemId(), null));
   }
 
+  public CompletableFuture<Result<MultipleRecords<Loan>>> findOpenLoansByUserIdWithItem(
+    LoanAndRelatedRecords loanAndRelatedRecords) {
+    String userId = loanAndRelatedRecords.getLoan().getUser().getId();
+    final Result<CqlQuery> statusQuery = getStatusCQLQuery("Open");
+    final Result<CqlQuery> userIdQuery = exactMatch("userId", userId);
+    Result<CqlQuery> cqlQueryResult = statusQuery
+      .combine(userIdQuery, CqlQuery::and);
+
+    return queryLoanStorage(Integer.MAX_VALUE, cqlQueryResult)
```

Comment:
Processing up to 2147483647 in memory will cause an out-of-memory error of this module.
Lateron a materialTypeId filter and an loanTypeId fitler is applied, can these filters been included in the CQL query?
Or can we replace the CQL query by an SQL count(*) query that does not return the loans but only the loan count?

Reply:
@julianladisch, what value would you suggest using for limit? The value was not defined in the requirements.
I cannot include a materialTypeId and an loanTypeId in the CQL because loans table doesn't have them.  And the tables do not have relations.
I don't quite understand the approach with SQL. We prepare CQL to call circulation-storage module, how can we do it with SQL? Could you please clarify it or maybe provide some example?

Reply:
You may process the open loans in batches of 1000 loans until all loans have been processed. For 2500 open loans you need 3 batches from the paging endpoint (1000 + 1000 + 500).

This query works for me at the `/loan-storage/loans` API:
`userId==16757efe-86ac-40bb-bdd6-fafee02463c7 AND status.name==Open AND loanPolicyId==d9cd0bed-1b49-4b5e-a7bd-064b8d177231`

It is true that the materialTypeId requires an additional subquery. It should be considered to save the materialTypeId in the loan dataset, similar to the loanTypeId, to improve performance. However, this is a separate task out of scope of this issue.

For the item limit we do not need the loan records, we only need the loan count grouped by loan type and material type. For best response times mod-circulation-storage should provide a special API endpoint that provides these numbers (and does not transmit all the loan records) and uses a dedicated count(*) ... GROUP BY SQL query. This is probably out of scope of this issue.

Have you tested how this pull request degrades performance?

Reply:
@julianladisch  Why is the loan policy relevant for that query?

According to Holly's answer to my clarifying question on the issue, the loan policy *should not* be considered when determining which loans contribute to the item limit calculation.

Have I misunderstood something?

Reply:
@julianladisch, I've implemented pagination logic

Reply:
@marcjohnson-kint The loan policy is relevant for the simple solution only.

Reply:
@roman-barannyk I'll answer on the other conversion.